### ISSUE TITLE
🐛 BUG: Astro.resolve() not working as expected (astro@0.21.0-next.3)

### ISSUE THREAD

**Issue by bobaekang** [snowpackjs/astro#1821]
### What version of `astro` are you using?

0.21.0-next.3

### What package manager are you using?

npm

### What operating system are you using?

StackBlitz for the example, Windows (WSL 1) for my local use

### Describe the Bug

This Issue is sort of a follow-up to #1715.

There are two aspects of the erroneous `Astro.resolve()` behavior:

1. Using `Astro.resolve()` to import  CSS stylesheet seems to generate an unused asset that duplicates the one that's in use in the build output.
2. Using `Astro.resolve()` to import JS script still fails because the build output includes neither the asset nor the valid path for the (missing) asset.

Here's an example source:

```astro
---
---
<html lang="en">

<head>
	<meta charset="utf-8" />
	<link rel="icon" type="image/x-icon" href="/favicon.ico" />
	<meta name="viewport" content="width=device-width" />

	<!-- THIS WORKS, BUT HAS A DUPLICATE STYLESHEET IN BUILD OUTPUT -->
	<link rel="stylesheet" href={Astro.resolve('../style.css')} type="text/css" />

	<!-- THIS DOESN'T WORK -->
	<script src={Astro.resolve('../script.js')} />

	<title>Welcome to Astro</title>
</head>

<body>
	<h1>Welcome to <a href="https://astro.build/">Astro</a></h1>
</body>

</html>
```

And here's the build output:

```html
<html lang="en"><head>
	<meta charset="utf-8">
	<link rel="icon" type="image/x-icon" href="/favicon.ico">
	<meta name="viewport" content="width=device-width">

	<!-- THIS WORKS, BUT HAS A DUPLICATE STYLESHEET IN BUILD OUTPUT -->
	<link rel="stylesheet" href="assets/index.7ce926d6.css" type="text/css">

	<!-- THIS DOESN'T WORK -->
	<script src="/home/projects/github-4sqhr1/src/script.js"></script>

	<title>Welcome to Astro</title>
</head>

<body>
	<h1>Welcome to <a href="https://astro.build/">Astro</a></h1>
</body></html>
```

Here's a screenshot comparing duplicated stylesheets (left in use, right not in use):

![image](https://user-images.githubusercontent.com/22449454/141659375-6beb0452-c6a3-451c-9810-84522fffbc11.png)

In contrast, `Astro.resolve()` works with dev server.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-4sqhr1 (includes build output `/dist`)

**Comment by natemoo-re** [comment]
I just ran into this. It looks like we're only transforming `script[src]` attributes if the `hoist` attribute is present, which is incorrect.

Not sure what's going on with the unused CSS.

**Comment by bobaekang** [comment]
@natemoo-re Unfortunately, it doesn't look like adding `hoist` changes much.

Source `index.astro`:
```astro
---
---
<html lang="en">

<head>
	<meta charset="utf-8" />
	<link rel="icon" type="image/x-icon" href="/favicon.ico" />
	<meta name="viewport" content="width=device-width" />

	<!-- THIS WORKS, BUT HAS A DUPLICATE STYLESHEET IN BUILD OUTPUT -->
	<link rel="stylesheet" href={Astro.resolve('../style.css')} type="text/css" />

	<!-- THIS DOESN'T WORK -->
	<script src={Astro.resolve('../script.js')} hoist />

	<title>Welcome to Astro</title>
</head>

<body>
	<h1>Welcome to <a href="https://astro.build/">Astro</a></h1>
</body>

</html>
```

Build output:
```html
<html lang="en"><head>
	<meta charset="utf-8">
	<link rel="icon" type="image/x-icon" href="/favicon.ico">
	<meta name="viewport" content="width=device-width">

	<!-- THIS WORKS, BUT HAS A DUPLICATE STYLESHEET IN BUILD OUTPUT -->
	<link rel="stylesheet" href="assets/index.7ce926d6.css" type="text/css">

	<!-- THIS DOESN'T WORK -->
	

	<title>Welcome to Astro</title>
<script src="/home/projects/github-4sqhr1-wx3qjb/src/script.js" hoist="true" astro-script="//script-0"></script></head>

<body>
	<h1>Welcome to <a href="https://astro.build/">Astro</a></h1>
</body></html>
```

Here's the StackBlitz link: https://stackblitz.com/edit/github-4sqhr1-wx3qjb

**Comment by matthewp** [comment]
I'm going to take a look, thanks for the reproduction!

**Comment by matthewp** [comment]
Note that a fix for hoisted scripts is coming in the next `next` release (today), so specifically curious about the double CSS issue you're seeing.

**Comment by matthewp** [comment]
Yeah, I can see that there are 2 CSS files being generated, but only one being loaded. Very interesting, but I might have an idea of what that is, so I'm going to debug it. My suspicious is that we are emitting a pre-minify version.

**Comment by matthewp** [comment]
This was fixed in #1838 and will soon be available in the next `next` release.
